src/controllers/connectionManager.ts

```diff
@@ -1249,25 +1249,15 @@ export default class ConnectionManager {
     }
 
     /**
-     * create a new connection with the connectionCreds provided
-     * @param fileUri
-     * @param connectionCreds ConnectionInfo to connect with
-     * @param promise Deferred promise to resolve when connection is complete
-     * @returns true if connection is successful, false otherwise
+     * Handles password-based credential authentication by prompting for password if needed.
+     * This method checks if a password is required and prompts the user if it's not saved or available.
+     *
+     * @param connectionCreds The connection credentials to process
+     * @returns Promise that resolves to true if password handling was successful, false if user cancelled
      */
-    public async connect(
-        fileUri: string,
+    public async handlePasswordBasedCredentials(
```

Comment:
Not sure why git decides to show the diff this way. The change here is to extract some code for adding pwd to the cred that Aasim added a few weeks ago into a helper function and this function is later used by connectTool.

Reply:
For example I have localhost master saved, now I want to connect to localhost TestDB2, I'll re-use the master profile, get the pwd, plug in the TestDB2 name, attempt the connection and see if the connection succeeds or not.